src/lib/db/environment-store.ts

```diff
@@ -112,6 +156,54 @@ export default class EnvironmentStore implements IEnvironmentStore {
         return rows.map(mapRow);
     }
 
+    async getAllWithCounts(query?: Object): Promise<IEnvironment[]> {
+        let qB = this.db<IEnvironmentsWithCountsTable>(TABLE)
+            .select(
+                '*',
+                this.db.raw(
+                    '(SELECT COUNT(*) FROM project_environments WHERE project_environments.environment_name = environments.name) as project_count',
+                ),
+                this.db.raw(
+                    '(SELECT COUNT(*) FROM api_tokens WHERE api_tokens.environment = environments.name) as api_token_count',
+                ),
+                this.db.raw(
+                    '(SELECT COUNT(*) FROM feature_environments WHERE enabled=true AND feature_environments.environment = environments.name) as enabled_toggle_count',
+                ),
+            )
+            .orderBy([
+                { column: 'sort_order', order: 'asc' },
+                { column: 'created_at', order: 'asc' },
+            ]);
+        if (query) {
+            qB = qB.where(query);
+        }
+        const rows = await qB;
+        return rows.map(mapRowWithCounts);
+    }
+
+    async getProjectEnvironments(
+        projectId: string,
+    ): Promise<IProjectEnvironment[]> {
+        let qB = this.db<IEnvironmentsWithProjectCountsTable>(TABLE)
+            .select(
+                '*',
+                this.db.raw(
+                    '(SELECT COUNT(*) FROM api_tokens LEFT JOIN api_token_project ON api_tokens.secret = api_token_project.secret WHERE api_tokens.environment = environments.name AND (project = :projectId OR project IS null)) as project_api_token_count',
```

Comment:
I _think_ this is okay, but the left join in a sub query worries me a bit

@chriswk thoughts here? It's a rarely run query and it only affects the UI

Reply:
Okay we spoke, we think this is okay